core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -0,0 +1,360 @@
+      1,
+      1,
+      servers,
+      servers.head.consumerCoordinator.offsetsTopicConfigs)
+    // create the test topic with all the brokers as replicas
+    TestUtils.createTopic(this.zkClient, topic, 1, 1, this.servers)
+  }
+
+  @After
+  override def tearDown() = {
+    removeAllAcls
+    super.tearDown()
+  }
+
+  @Test
+  def testAuthorization() {
+    val socket = new Socket("localhost", servers.head.boundPort())
+
+    for ((key, request) <- RequestKeyToRequest) {
+
+      removeAllAcls
+
+      val buffer = ByteBuffer.allocate(request.sizeInBytes)
+      request.writeTo(buffer)
+      buffer.rewind()
+      val requestBytes = buffer.array()
+
+      sendRequestAndVerifyResponseErrorCode(socket, key, requestBytes, ErrorMapping.AuthorizationCode)
+
+      for ((resource, acls) <- RequestKeysToAcls(key))
+        addAndVerifyAcls(acls, resource)
+
+      sendRequestAndVerifyResponseErrorCode(socket, key, requestBytes, ErrorMapping.NoError)
+    }
+  }
+
+    @Test
+    def testProduceNeedsAuthorization() {
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Describe)), topicResource)
+      try {
+        sendRecords(numRecords, tp)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: ApiException => Assert.assertEquals(Errors.AUTHORIZATION_FAILED.exception().getMessage, e.getMessage)
+      }
+    }
+
+    @Test
+    def testOnlyWritePermissionAllowsWritingToProducer() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      try {
+        sendRecords(numRecords, tp)
+      } catch {
+        case e: Exception => Assert.fail("Should not fail")
+      }
+    }
+
+    @Test
+    def testCreatePermissionNeededForWritingToNonExistentTopic() {
+      val newTopic = "newTopic"
+      val topicPartition = new TopicPartition(newTopic, 0)
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), new Resource(Topic, newTopic))
+      try {
+        sendRecords(numRecords, topicPartition)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: TimeoutException =>
+        //TODO Need to update the producer so it actually throws the server side of exception.
+        case e: Exception => Assert.fail("Only timeout exception should be thrown.")
+      }
+
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Create)), Resource.ClusterResource)
+      try {
+        sendRecords(numRecords, topicPartition)
+      } catch {
+        case e: Exception => Assert.fail("No exception should be thrown.")
+      }
+    }
+
+    @Test
+    def testConsumerNeedsAuthorization() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      //TODO: Ideally we would want to test that when consumerGroup permission is not present we still get an AuthorizationException
```

Comment:
That JIRA is just about making sure that the consumer doesn't block. It is likely to throw a TimeoutException once it's fixed. We should have a JIRA that depends on KAFKA-2391 for the authorization part.